pkg/library/container/container.go

```diff
@@ -46,7 +48,7 @@ import (
 // rewritten as Volumes are added to PodAdditions, in order to support e.g. using one PVC to hold all volumes
 //
 // Note: Requires DevWorkspace to be flattened (i.e. the DevWorkspace contains no Parent or Components of type Plugin)
-func GetKubeContainersFromDevfile(workspace *dw.DevWorkspaceTemplateSpec, securityContext *corev1.SecurityContext, pullPolicy string) (*v1alpha1.PodAdditions, error) {
+func GetKubeContainersFromDevfile(k8sClient client.Client, namespace string, workspace *dw.DevWorkspaceTemplateSpec, securityContext *corev1.SecurityContext, pullPolicy string) (*v1alpha1.PodAdditions, error) {
```

Comment:
I'm mildly concerned with the high-level changes to `pkg/library/container` -- this package was intended to provide a straight conversion from devfile resources to Kubernetes resources, so including a client and depending on cluster state is something I'd prefer to avoid if possible. 

However, I don't see a great way of supporting this functionality cleanly otherwise. Probably the best solution is some "convert-to-container config" struct that sets whether we set a CPU limit by default, but that's also a much bigger change.